src/app/config.py

```diff
@@ -0,0 +1,161 @@
+    FIRESTORE_EMULATOR_HOST: str
+    STORAGE_EMULATOR_HOST: str
+
+
+class CloudDevelopmentConfig(Config):
+    def __init__(self):
+        super().__init__()
+        self.SCHEMA_BUCKET_NAME = get_value_from_env("SCHEMA_BUCKET_NAME")
+        self.GOOGLE_APPLICATION_CREDENTIALS = get_value_from_env(
+            "GOOGLE_APPLICATION_CREDENTIALS"
+        )
+
+    SCHEMA_BUCKET_NAME: str
+    GOOGLE_APPLICATION_CREDENTIALS: str
+
+
+class TestingConfig(Config):
+    def __init__(self):
+        super().__init__()
+        self.TEST_DATASET_PATH = get_value_from_env("TEST_DATASET_PATH")
+        self.TEST_SCHEMA_PATH = get_value_from_env("TEST_SCHEMA_PATH")
+
+    TEST_DATASET_PATH: str
+    TEST_SCHEMA_PATH: str
+
+
+class IntegrationTestingRemoteCloudConfig(TestingConfig):
+    def __init__(self):
+        super().__init__()
+        self.API_URL = get_value_from_env("API_URL")
+        self.SCHEMA_BUCKET_NAME = get_value_from_env("SCHEMA_BUCKET_NAME")
+
+    SCHEMA_BUCKET_NAME: str
+    API_URL: str
+
+
+class IntegrationTestingLocalCloudConfig(TestingConfig):
+    def __init__(self):
+        super().__init__()
+        self.API_URL = get_value_from_env("API_URL")
+        self.GOOGLE_APPLICATION_CREDENTIALS = get_value_from_env(
+            "GOOGLE_APPLICATION_CREDENTIALS"
+        )
+
+    API_URL: str
+    GOOGLE_APPLICATION_CREDENTIALS: str
+
+
+class IntegrationTestingLocalSDSConfig(TestingConfig):
+    def __init__(self):
+        super().__init__()
+        self.GOOGLE_APPLICATION_CREDENTIALS = get_value_from_env(
+            "GOOGLE_APPLICATION_CREDENTIALS"
+        )
+        self.SCHEMA_BUCKET_NAME = get_value_from_env("SCHEMA_BUCKET_NAME")
+
+    SCHEMA_BUCKET_NAME: str
+    GOOGLE_APPLICATION_CREDENTIALS: str
+
+
+class IntegrationTestingLocalConfig(TestingConfig):
+    def __init__(self):
+        super().__init__()
+        self.FIRESTORE_EMULATOR_HOST = get_value_from_env("FIRESTORE_EMULATOR_HOST")
+        self.STORAGE_EMULATOR_HOST = get_value_from_env("STORAGE_EMULATOR_HOST")
+
+    FIRESTORE_EMULATOR_HOST: str
+    STORAGE_EMULATOR_HOST: str
+
+
+class UnitTestingConfig(TestingConfig):
+    def __init__(self):
+        super().__init__()
+        self.SCHEMA_BUCKET_NAME = get_value_from_env("SCHEMA_BUCKET_NAME")
+
+    SCHEMA_BUCKET_NAME: str
+
+
+match CONF:
+    case "docker-dev":
+        config = ServiceEmulatorDevelopementConfig()
+    case "cloud-dev":
+        config = CloudDevelopmentConfig()
+    case "IntegrationTestingLocalSDS":
+        config = IntegrationTestingLocalSDSConfig()
+    case "IntegrationTestingCloud":
+        config = IntegrationTestingLocalCloudConfig()
+    case "IntegrationTestingDocker":
+        config = IntegrationTestingLocalConfig()
+    case "unit":
+        config = UnitTestingConfig()
+    case "cloud-build":
+        config = CloudBuildConfig()
+    case "cloud-integration-test":
+        config = IntegrationTestingRemoteCloudConfig()
+    case "default":
+        config = Config()
```

Comment:
I'd argue you should put this into its own factory class, this file is getting pretty large as is.